pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -283,6 +283,9 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundHTTPListenerOptsForPort
 			},
 			ServerName:          EnvoyServerName,
 			DelayedCloseTimeout: features.DelayedCloseTimeout,
+			Http2ProtocolOptions: &core.Http2ProtocolOptions{
```

Comment:
do we need it on outbound as well?

Reply:
and gateway

Reply:
@zhiminxiang this will make https://github.com/knative/serving/issues/7933#issuecomment-786139169 not required I think

Reply:
From [these docs](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/upgrades#websocket-over-http-2-or-http-3-hops):

> This is accomplished via Extended CONNECT (RFC8441) support, turned on by setting allow_connect true at the second layer Envoy.

I took this to mean that the inbound listeners need this set but not the outbound although I'm not entirely clear on what "second layer" referred to there.

And yeah I think this should be added to gateways